File: DSA07024.py
# Test sai
from collections import deque

def function():
    n = int(input())
    ls = list(map(int, input().split()))
    ls += [0]
    stack = deque([(ls[0], 0)])
    result = 0
    for index, i in enumerate(ls[1:], start=1):
        k = index
        while stack and stack[-1][0] >= i:
            val, pos = stack.pop()
            if index - pos >= val:
                result = max(result, val)
            k = pos
        stack.append((i, k))
    print(result)


if __name__ == "__main__":
    for i in range(int(input())):
        function()

File: DSA07024.cpp
#include <iostream>
#include <vector>
#include <stack>

using namespace std;

void function() {
    int n;cin >> n;
    vector<int> ls(n);
    for (auto& x : ls) cin >> x;
    ls.push_back(0);

    stack<pair<int, int>> st;
    int result = 0;
    for (int index = 0; index < n + 1; ++index) {
        int i = ls[index];
        int k = index;
        while (!st.empty() && st.top().first >= i) {
            auto [val, pos] = st.top();
            if (index - pos >= val)
                result = max(result, val);
            k = pos;
            st.pop();
        }
        st.push({i, k});
    }
    cout << result << endl;
}

int main() {
    int t;
    cin >> t;
    while (t--) {
        function();
    }
    return 0;
}
